### ISSUE TITLE
Firefox: Dependency on KeyboardEvent.initKeyEvent causes auto-fill breakage in Nightly

### ISSUE THREAD

**Issue by denschub** [bitwarden/browser#2000]
## Describe the Bug

In `src/content/autofill.js`, there is a dependency on `KeyboardEvent.initKeyEvent` for Firefox:

https://github.com/bitwarden/browser/blob/ef4dddcef3115e973b133d0b473c315425ca43b4/src/content/autofill.js#L841-L844

Given this was a Firefox-only thing and it's been causing web compatibility issues, [it was removed yesterday](https://bugzilla.mozilla.org/show_bug.cgi?id=1717760), shipping in today's Nightly. Unfortunately, this breaks Bitwarden's auto-fill, as the codepath above only checks for `isFirefox`, and does not check if `initKeyEvent` is defined or not.

## Steps To Reproduce

1. Upgrade to Firefox Nightly 2021-08-10 or newer.
2. Try using the Auto-Fill keyboard shortcut.

## Expected Result

Auto-filling fields.

## Actual Result

Nothing happens. 

## Environment

- Browser: Firefox Nightly 93a1, >= 2021-08-10
- Build Version (go to "Settings" → "About" in the app): 20210810094908

**Comment by streetwolf** [comment]
It appears this change is by design on Nightly and in all probability will make it's way to a stable version.  That being said can you please update BW to account for this change so that it will work on Nightly and the future stable version.

**Comment by masayuki-nakano** [comment]
https://github.com/bitwarden/browser/blob/ef4dddcef3115e973b133d0b473c315425ca43b4/src/content/autofill.js#L838-L856

So, it should do:
```
function normalizeEvent(el, eventName) {
  return new KeyboardEvent(eventName, {
    bubbles: true,
    view: null,
    keyCode: 0,
    charCode: 0,
    which: 0,
  });
}
```

**Comment by MGibson1** [comment]
This fix just missed our planned August release, but will make it in the next. Next release will still be in time to preempt this nightly change from hitting beta/stable.

**Comment by umerf52** [comment]
> 
> 
> This fix just missed our planned August release, but will make it in the next. Next release will still be in time to preempt this nightly change from hitting beta/stable.

When is the next planned Bitwarden release supposed to be available?

**Comment by K900** [comment]
Is there any chance to get this backported to the current version? Building from source requires manually reloading the extension every time...

**Comment by jonathonf** [comment]
If you set `xpinstall.signatures.required=false` you can install unsigned extensions (whether one you built from source or grabbed from the pipeline build).

We're running on Nightly so breakages should be expected, and extension developers aren't likely to worry too much about such a rapidly moving target (it could all change again).

Devs have already said there will be a new Bitwarden extension release before the next Beta is due, so there's a near timescale.

**Comment by sciguyryan** [comment]
A push of a version that fixes this would be greatly appreciated. It is quite inconvenient that autofill is inoperable and it does brake a significant function within the addon.

Thanks for fixing this so quickly.

**Comment by jonathonf** [comment]
There is already a fix available: https://github.com/bitwarden/browser/pull/2001#issuecomment-897031836

**Comment by shawnz** [comment]
Note that you can also set `dom.keyboardevent.init_key_event.enabled` to "true" to allow the release version to keep working for the time being.

**Comment by Mte90** [comment]
When there will be published the new version? I see that there is a 1.52.1 but on marketplace there is still the 1.52.0.

**Comment by masayuki-nakano** [comment]
FYI: Firefox puts off unshipping it only in addons.
